pulsar-common/src/main/java/org/apache/pulsar/common/util/BundleDataComparator.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.common.util;
+
+import java.io.Serializable;
+import java.util.Comparator;
+import org.apache.pulsar.policies.data.loadbalancer.BundleData;
+
+public class BundleDataComparator implements Comparator<BundleData>, Serializable {
```

Comment:
Can we implement the `Comparable<BundleData>` interface for `BundleData` class?

Reply:
If we implement the Comparable<BundleData> interface for BundleData class, we have to override the `equals` and `hashcode` method for BundleData, or bug spot tools will report errors like:
```
[ERROR] Medium: org.apache.pulsar.policies.data.loadbalancer.BundleData defines compareTo(BundleData) and uses Object.equals() [org.apache.pulsar.policies.data.loadbalancer.BundleData] At BundleData.java:[lines 103-107] EQ_COMPARETO_USE_OBJECT_EQUALS
```
But i am afraid that it will influence logic in the other places.

Reply:
I don't think overriding the `equals` and `hashcode` methods will influence other places. Can you confirm it?

Reply:
revert to implement the `Comparable` interface.